pexpect/pty_spawn.py

```diff
@@ -430,11 +452,7 @@ def read_nonblocking(self, size=1, timeout=-1):
         # For this case, I test isalive() before doing any reading.
         # If isalive() is false, then I pretend that this is the same as EOF.
```

Comment:
In general, on EOF, select will indicate that there is something to be read, and then - depending on the system - read will either fail or return an empty string. This comment appears to be saying that on Solaris, read on a pty after EOF will block, so we can only detect EOF by checking if the child process has gone. If so, by reading before this check, we could get blocked.

It may be that that is outdated, and Solaris has since been fixed to do something sane. But I don't want to throw away hard won compatibility code until we're confident that it's no longer needed. @jquast has been setting up a greater diversity of different Unix systems than I have ready access to - he may be able to check it.